contrib/internal/httptrace/httptrace.go

```diff
@@ -55,3 +81,69 @@ func FinishRequestSpan(s tracer.Span, status int, opts ...tracer.FinishOption) {
 	}
 	s.Finish(opts...)
 }
+
+// Helper function to return the IP network out of a string.
+func ippref(s string) *netaddr.IPPrefix {
+	if prefix, err := netaddr.ParseIPPrefix(s); err == nil {
+		return &prefix
+	}
+	return nil
+}
+
+// getClientIP uses the request headers to resolve the client IP. If a specific header to check is provided through
+// DD_CLIENT_IP_HEADER, then only this header is checked.
```

Comment:
This describes how the code in this file works, not what the function being defined does. So maybe try:
```suggestion
// getClientIP attempts to find the client IP address in the given request r.
```